tensorflow/core/util/tensor_format.h

```diff
@@ -574,9 +583,17 @@ inline TensorShape ShapeFromFilterTensorFormat(FilterTensorFormat format,
 }
 
 // Return a tensor shape of the specified 'format', and dimensions.
+inline Status ShapeFromFormatWithStatus(TensorFormat format, int64_t N, int64_t H,
+                                         int64_t W, int64_t C, TensorShape *shape) {
+  return ShapeFromFormatWithStatus(format, N, {H, W}, C, shape);
+}
+
 inline TensorShape ShapeFromFormat(TensorFormat format, int64_t N, int64_t H,
                                    int64_t W, int64_t C) {
-  return ShapeFromFormat(format, N, {H, W}, C);
+  TensorShape shape;
+  Status status = ShapeFromFormatWithStatus(format, N, H, W, C, &shape);
+  CHECK(status.ok());
```

Comment:
We want to avoid these kinds of `CHECK`s in general.  Ideally we would eliminate all uses of `ShapeFromFormat`, instead using your new `Status` version..  There doesn't seem to be _too_ many usages in TensorFlow, but I'm not sure how large of a change it would be to eliminate this entirely.  Can you investigate?

Reply:
Let's remove this, and see what the damage is.